tools/mksurfdata_map/src/mkgridmapMod.F90

```diff
@@ -487,12 +487,14 @@ subroutine gridmap_check(gridmap, mask_src, frac_dst, caller)
     
     sum_area_i = 0.0_r8
     do ni = 1,ns_i
-       sum_area_i = sum_area_i + gridmap%area_src(ni)*mask_src(ni)*re**2
+!      sum_area_i = sum_area_i + gridmap%area_src(ni)*mask_src(ni)*re**2
+       sum_area_i = sum_area_i + gridmap%area_src(ni)*gridmap%frac_src(ni)*re**2
     enddo
 
     sum_area_o = 0.
     do no = 1,ns_o
-       sum_area_o = sum_area_o + gridmap%area_dst(no)*frac_dst(no)*re**2
+!      sum_area_o = sum_area_o + gridmap%area_dst(no)*frac_dst(no)*re**2
+       sum_area_o = sum_area_o + gridmap%area_dst(no)*gridmap%frac_dst(no)*re**2
```

Comment:
I had to reverse changes in the above few lines to get test 007 to PASS, as explained in my last commit. 

Earlier testing with the f09 resolution had been passing, thereby convincing me that the code was correct. The 007 test is for the 10x15 resolution and was not passing due to increased sensitivity to the fact that mask_src is either 0 or 1, while gridmap%frac_src is a fraction.

I am open to discussing alternative solutions.

Reply:
@billsacks I read your message to Bob Oehmke. The gist of it is that gridmap%frac_src and gridmap%frac_dst should equal 1 everywhere in our new map_ files in which the corresponding masks equal 1 everywhere. This means that the code as written before this commit uncovered a problem with the specific map_ file, and I should change the code back again. I will wait for confirmation of this.

Reply:
@billsacks , sorry, I am revisiting my last post and reconsidering the statement about changing the code back again. Correcting the map_ file to have gridmap%frac_src and gridmap%frac_dst equal to 1 everywhere will not correct the error that I was getting in test 007 that made me reverse the changes in the above few lines.